### DISCUSSION TITLE
DNS verification for ACME plugin

### DISCUSSION THREAD

**Discussion by tiny-pangolin** [Kong/kong#13879] [Ideas and feature requests]
I would like to be able to issue certificates using the ACME plugin with DNS verification so I can issue wildcard certs and/or issue well known certs to internal services

**Comment by xianghai2** [comment]
Hi @tiny-pangolin , could you clarify your use case in details? thanks.

**Comment by tiny-pangolin** [comment]
my current use case is I run multiple internal sites using docker and would like to issue ACME certificates to those internal services. Since they are not exposed publically I can't use the acme plugin since it only supports http verification, and AFAIK there is not something like cert-manager that is used in docker so it usually falls on the apigateway or reverse proxy to issue certs in docker. The reason for not issuing internal certs is that it is a giant pain to add CA certs on mobile devices and multiple desktop operating systems

**Reply by xianghai2** [reply]
The ACME plugin should work with both 'https' and 'http' protocol. 
https://docs.konghq.com/hub/kong-inc/acme/configuration/

Looks like that I may not catch the point of this configuation issue?

**Reply by tiny-pangolin** [reply]
The issue with this plugin for my use case is that HTTP verification requires port 80 and/or 443 is publically accessible. I was hoping Kong could add[ DNS-01 verification](https://letsencrypt.org/docs/challenge-types/#dns-01-challenge) so I could get a widely accepted certificates without publically exposing my api gateway

**Reply by tiny-pangolin** [reply]
Currently I'm using traefik for this but, I would like to evaluate kong for improved performance and the ability to manage API keys

**Comment by xianghai2** [comment]
Got it. 
At this moment, acme plugin does not support dns-01 challenge. It’s on our roadmap but not prioritized right now. thanks.
